### DISCUSSION TITLE
How to stop Whisper downloading the model each time

### DISCUSSION THREAD

**Discussion by Dec-35** [openai/whisper#1013] [Q&A] [answered]
I have to run whisper every single day and it downloads the model (medium in my case so pretty long download on slow internet) almost every time I run it. How can I stop that behaviour?

![image](https://user-images.githubusercontent.com/88966922/222226620-3f270570-8f8e-41a9-9313-86503fe3ca7f.png)

It also gets stuck on 19kiB/s whearas my internet is about 6 or 7 mB/s

**Comment by jongwook** [comment, marked as answer]
It should not download the model each time once it is fully downloaded. It appears that the Azure CDN that we're using is not particularly friendly with your network. Does it download any faster if you try this address in your browser:

https://openaipublic.azureedge.net/main/whisper/models/345ae4da62f9b3d59415adc60127b97c714f32e89e936602e85993674d08dcb1/medium.pt

Alternatively, you could try this non-CDN address which is typically slower:

https://openaipublic.blob.core.windows.net/main/whisper/models/345ae4da62f9b3d59415adc60127b97c714f32e89e936602e85993674d08dcb1/medium.pt

Once downloaded, Whisper can use the model if you copy them to `~/.cache/whisper/medium.pt`.

**Reply by pranabenator** [reply]
It does not take them from the cache but downloads again is my experience. Is there any way to explicitly instruct Whisper check the cache?

**Reply by phineas-pta** [reply]
you can download the checkpoint to a permanent directory, then load the model with `whisper.load_model("PATH TO .pt FILE")`

**Reply by lazarok09** [reply]
@phineas-pta  where is saved in the disk the particular model ?

**Reply by vincedani** [reply]
👋 I would like to follow the starting guidelines to use the project, but model loading stuck without any output:

```
$ pip install -U openai-whisper
$ python
>>> import whisper
>>> model = whisper.load_model("tiny.en") % never returns
```
The `load_model` somehow uses 100% of my CPU, and changing the device to `cuda` didn't help me move on. I've also tried to download the model and replace the `tiny.en` to the specific path, however, `load_model` blocks the execution for hours.

Do you have any ideas to solve the issue?

**Reply by skywalk-explore** [reply]
do you have a link the to the small model that can be used on Mac ? The medium model is a bit large and takes too long to process files.

**Reply by glangford** [reply]
> do you have a link the to the small model that can be used on Mac ?

Links to all the models can be found here in the whisper source:
https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/__init__.py#L17

**Reply by skywalk-explore** [reply]
Thanks man.
